libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -168,6 +170,12 @@ method readOnce*(s: LPChannel,
   ## channels are blocked - in particular, this means that reading from one
   ## channel must not be done from within a callback / read handler of another
   ## or the reads will lock each other.
+  if s.connDown:
+    raise newLPStreamConnDownError()
+  if s.remoteReset:
+    raise newLPStreamResetError()
+  if s.atEof():
+    raise newLPStreamRemoteClosedError()
   try:
```

Comment:
We should remap EOF to a `newLPStreamConnDownError` on this try

Reply:
Of course, but since `readOnce` will block until bytes are available, the EOF may happen during this waiting time